crates/sui-graphql-client/src/lib.rs

```diff
@@ -539,6 +606,139 @@ impl Client {
+            .and_then(|d| d.object)
+            .and_then(|o| o.dynamic_field)
+            .map(|df| df.into());
+        Ok(result)
+    }
+
+    /// Access a dynamic object field on an object using its name. Names are arbitrary Move values whose
+    /// type have copy, drop, and store, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// The `name` argument can be either a [`BcsName`] for passing raw bcs bytes or a type that
+    /// implements Serialize.
+    ///
+    /// This returns [`DynamicFieldOutput`] which contains the name, the value as json, and object.
+    pub async fn dynamic_object_field(
+        &self,
+        address: Address,
+        type_: &str,
+        name: impl Into<NameValue>,
+    ) -> Result<Option<DynamicFieldOutput>, Error> {
+        let bcs = name.into().0;
+        let operation = DynamicObjectFieldQuery::build(DynamicFieldArgs {
+            address,
+            name: crate::query_types::DynamicFieldName {
+                type_: type_.to_string(),
+                bcs: crate::query_types::Base64(base64ct::Base64::encode_string(&bcs)),
+            },
+        });
+
+        let response = self.run_query(&operation).await?;
+
+        if let Some(errors) = response.errors {
+            return Err(Error::msg(format!("{:?}", errors)));
+        }
+
+        let result: Option<DynamicFieldOutput> = response
+            .data
+            .and_then(|d| d.object)
+            .and_then(|o| o.dynamic_object_field)
+            .map(|df| df.into());
+        Ok(result)
+    }
+
+    /// Get a page of dynamic fields for the provided address. Note that this will also fetch
+    /// dynamic fields on wrapped objects.
+    ///
+    /// This returns [`Page`] of [`DynamicFieldOutput`]s.
+    pub async fn dynamic_fields(
+        &self,
+        address: Address,
+        after: Option<String>,
+        before: Option<String>,
+        first: Option<i32>,
+        last: Option<i32>,
+    ) -> Result<Option<Page<DynamicFieldOutput>>, Error> {
+        let operation = DynamicFieldsOwnerQuery::build(DynamicFieldConnectionArgs {
+            address,
+            after,
+            before,
+            first,
+            last,
+        });
+        let response = self.run_query(&operation).await?;
+
+        if let Some(errors) = response.errors {
+            return Err(Error::msg(format!("{:?}", errors)));
+        }
+
+        if let Some(dfs) = response.data {
+            if let Some(owner) = dfs.owner {
+                let page_info = owner.dynamic_fields.page_info;
+                let nodes = owner.dynamic_fields.nodes;
+                let jsons = nodes
+                    .into_iter()
+                    .map(|df| df.into())
+                    .collect::<Vec<DynamicFieldOutput>>();
+
+                Ok(Some(Page::new(page_info, jsons)))
+            } else {
+                Ok(None)
+            }
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
`let ... else` + nested patterns:

```suggestion
        let Some(Query { owner: Some(Owner { dynamic_fields }) }) = response.data else {
            return Ok(Page::empty());
        };
        
        Ok(Page::new(
          dynamic_fields.page_info,
          dynamic_fields
              .nodes
              .into_iter()
              .map(Into::into)
              .collect()))
    }
```